src/10-deep-strong-coupling.md

```diff
@@ -0,0 +1,1132 @@
+
+```python
+bra_labels, ket_labels = make_braket_labels(nm_list)
+```
+
+Our initial TLS ground state in vaccuum is the first state in the list:
+
+```python
+nm_list[0]
+```
+
+```python
+psi0 = basis(len(nm_list), 0)
+```
+
+Now for the Hamiltonian.
+
+```python
+H =  DeltaE*two_state + omega*bosons + U*interaction
+```
+
+```python
+times = np.linspace(0.0, 1*(2*np.pi/omega), 2000)
+```
+
+As we've done before, we can extract just the probabilities for states instead of the wavefunction by using projection opperators.
+
+```python
+projection_operators = [basis(len(nm_list), i).proj() for i in range(len(nm_list))]
+```
+
+We'll also extract expectation values for some of the Hamiltonian operators.
+
+```python
+result = sesolve(H, psi0, times, e_ops = projection_operators + [number, U*interaction, omega*bosons])
+
+probs = result.expect[:-3]      # Remove the last 3 expectations to get the state probabilities
+expected = result.expect[-3:]   # The last 3 expectations are the number, interaction and field energies
+```
+
+```python
+plot_sim(times/(2*np.pi/omega), probs ,labels=ket_labels)
+plt.title(f" {H_latex}   \n $\Delta E={DeltaE}$, $\omega={omega}$, $U={U}$, $\Psi_0=|0,->$     (Fig. 1)");
+plt.xlabel("Time ($2\pi/\omega$)");
+```
+
+Fig. 1 shows us that our picture is correct. The vacuum state "pulls" the other state into almost 100% occupation probability on a timescale associated with the [Rabi frequency](https://en.wikipedia.org/wiki/Vacuum_Rabi_oscillation) $\Omega = \sqrt{4U^2 + \delta^2}$ where $\delta$ is often called the "detuning" which is just the difference in energies of the two states. 
+
+```python
+rabi_omega = np.sqrt(4*U**2 + 1**2)
+```
+
+```python
+rapi_period = 1/rabi_omega
+print(rapi_period)
+```
+
+How are the results of Fig. 1 possible? Where has the energy come from to create a boson from nothing? Recall that the TLS has zero energy so it's not from there!
+
+The only place it can come from is the interaction energy. Let's take a look at the expectation of the boson and interaction energy to resolve this.
+
+```python
+plt.figure(figsize=(10,8))
+plt.plot(times/(2*np.pi/omega), expected[2], label="$\hbar\omega(a^{{\dagger}}a +1/2)$ - bosons")
+plt.plot(times/(2*np.pi/omega), expected[1], label="$U( a^{{\dagger}} + a )\sigma_x$ - interaction")
+plt.ylabel("Expected energy")
+plt.xlabel("Time ($2\pi/\omega$)")
+plt.title(f" {H_latex}   \n $\Delta E={DeltaE}$, $\omega={omega}$, $U={U}$, $\Psi_0=|0,->$     (Fig. 2)");
+plt.legend(loc='right');
+plt.show();
+```
+
+Fig. 2 shows is that the boson energy is being "borrowed" from interaction energy which can take on a form of energy debt. We actually saw hints of this way back in tutorial 4 (Fig. 13), but now that the coupling is so much stronger we can see the effect more clearly.
+
+
+### Arbitrary number of coupled states
+
+
+What happens when we remove the artificial restriction of having only two states? How does our picture change?
```

Comment:
perhaps change to "only two states and add more TLSs"? Is that correct?

Reply:
Not quite. We're not adding more TLS, we've just got a single TLS. We're allowing more than 1 bosons. We can only ever simulate a finite number of states. We started with two states, |0,-> and |1,+>, but we can have more. |2,->, |3,+> when we have negative parity.

Reply:
But I think what might be helpful ti change the sentence to read

> What happens when we remove the artificial restriction of **simulating** only two states? How does our picture change?

Changing having to simulating.

Reply:
Yeah this is helpful. If you think it's appropriate I would also add the phrase "allowing more than one bosons."

Reply:
Yes, lovely. I like that. Thanks